packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts

```diff
@@ -166,7 +173,7 @@ export class DbSyncStakePoolProvider extends DbSyncProvider(RunnableModule) impl
   }
 
   public async queryStakePools(options: QueryStakePoolsArgs): Promise<Paginated<Cardano.StakePool>> {
-    const { filters, pagination, rewardsHistoryLimit } = options;
+    const { filters, pagination, rewardsHistoryLimit = REWARDS_HISTORY_LIMIT_DEFAULT } = options;
```

Comment:
This has impact on APY calculation as well, yes.

While speaking about this activity I was thinkin about change this [code snippet](https://github.com/input-output-hk/cardano-js-sdk/blob/90880cf1be445808b2ca39b3bcd4757bef0f9981/packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts#L219-L224):

From
```
    // Get stake pools rewards cached
    const poolRewards = await this.#cache.get(
      queryCacheKey(StakePoolsSubQuery.REWARDS, orderedResultHashIds, options),
      () => this.#builder.queryPoolRewards(orderedResultHashIds, this.#epochLength, rewardsHistoryLimit),
      UNLIMITED_CACHE_TTL
    );
```
to
```
    // Get stake pools rewards cached
    const poolRewards = await this.#cache.get(
      queryCacheKey(StakePoolsSubQuery.REWARDS, orderedResultHashIds, options),
      () => this.#builder.queryPoolRewards(orderedResultHashIds, this.#epochLength, typeof rewardsHistoryLimit === 'number' ? rewardsHistoryLimit : 0),
      UNLIMITED_CACHE_TTL
    );
```

Reply:
Probably the cache key, needs to be dependent on `rewardsHistoryLimit` as well.

Reply:
Because `rewardsHistoryLimit` is used across different places within the method - [here](https://github.com/input-output-hk/cardano-js-sdk/blob/e7084dc9fb0d99aaae4c723263c8afadf3d08b3e/packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts#L223), [here](https://github.com/input-output-hk/cardano-js-sdk/blob/e7084dc9fb0d99aaae4c723263c8afadf3d08b3e/packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts#L229) and [here](https://github.com/input-output-hk/cardano-js-sdk/blob/e7084dc9fb0d99aaae4c723263c8afadf3d08b3e/packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts#L237), I decided to default it at the very beginning in the [options obj destruction](https://github.com/input-output-hk/cardano-js-sdk/blob/e7084dc9fb0d99aaae4c723263c8afadf3d08b3e/packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/DbSyncStakePoolProvider.ts#L176).
But if you see an issue with that I can change it to use inline checks and assignments following your example. @iccicci

Reply:
No @IvayloAndonov , I see no issues; mine was just the proposal for another approach (which actually I have not evaluated in the deeps). I'm fine with setting the default value to 3.